text/0000-std-rpc-api.md

```diff
@@ -0,0 +1,503 @@
+
+```cs
+Pong(somenumber, sometext); // Is this an RPC call?
+
+PongRpc(somenumber, sometext); // Is this a ServerRpc call or ClientRpc call?
+
+PongClientRpc(somenumber, sometext); // This is clearly a ClientRpc call
+```
+
+### Reliability
+
+RPCs are reliable by default which means they are guaranteed to be executed on the remote side. However, sometimes developers might want to opt-out reliability, which is often the case for non-critical events such as particle effects, sounds effects etc.
+
+Reliability configuration can be specified for both `ServerRpc` and `ClientRpc` methods at compile-time:
+
+```cs
+[ServerRpc]
+void MyReliableServerRpc() { /* ... */ }
+
+[ServerRpc(IsReliable = false)]
+void MyUnreliableServerRpc() { /* ... */ }
+
+[ClientRpc]
+void MyReliableClientRpc() { /* ... */ }
+
+[ClientRpc(IsReliable = false)]
+void MyUnreliableClientRpc() { /* ... */ }
+```
+
+Reliable RPCs will be received on the remote end in the same order as they are fired but this in-order guarantee only applies to RPCs on the same `NetworkObject`. Different `NetworkObject`s might have reliable RPCs called but executed in different order compared to each other. To put more simply, in-order reliable RPC execution is guaranteed per `NetworkObject` basis only.
+
+An RPC call made without an active connection will be dropped and will not be queued for send automatically. Both reliable and unreliable RPC calls have to be made when there is an active network connection established between a client and the server. Also reliable RPC calls made during connection will be dropped on disconnect as well.
+
+### Execution Table
+
+An RPC function **never** executes its body immediately since it's being a network construct. Even a `ServerRpc` called by a host (an instance that is a client and the server at the same time, aka listen server) will not be executed immediately but follow the regular network frame staging first.
+
+||Server|Client|Host (Server+Client)|
+|-:|:-:|:-:|:-:|
+|ServerRpc Network Send|❌|✅|✅|
+|ServerRpc Network Call|✅|❌|✅|
+|ServerRpc Direct Call|❌|❌|❌|
+|ClientRpc Network Send|✅|❌|✅|
+|ClientRpc Network Call|❌|✅|✅|
+|ClientRpc Direct Call|❌|❌|❌|
+
+## RPC Params
+
+Both `ServerRpc` and `ClientRpc` methods can be configured either by `[ServerRpc]`/`[ClientRpc]` attributes at compile-time and/or `ServerRpcParams`/`ClientRpcParams` at runtime.
+
+Developers can put `ServerRpcParams`/`ClientRpcParams` as the last parameter (optionally) and they could be used for a consolidated space for `XXXRpcReceiveParams` and `XXXRpcSendParams`.
+
+The network framework will inject `XXXRpcReceiveParams` when invoked by network receive handling (framework code) and will consume `XXXRpcSendParams` when invoked by RPC send call (user code).
```

Comment:
Suggest: The network framework will inject the corresponding XXXRpcReceiveParams to what the user had declared in code when invoked by network receive handling (framework code) and will consume XXXRpcSendParams when invoked by RPC send call (user code).